src/string.js

```diff
@@ -173,6 +173,38 @@ export function trimStart(str) {
   return (str + '_').trim().slice(0, -1);
 }
 
+/**
+ * Wrapper around String.replace that handles asynchronous resolution.
+ * @param {string} str
+ * @param {RegExp} regex
+ * @param {Function|string} replacer
+ * @return {!Promise<string>}
+ */
+export function asyncStringReplace(str, regex, replacer) {
+  const stringBuilder = [];
+  let lastIndex = 0;
+  if (typeof replacer === 'string') {
+    stringBuilder.push(Promise.resolve(str.replace(regex, replacer)));
```

Comment:
nit: no need to use stringBuilder here, I think the idea was just `return Promise.resolve(str.replace(...)))` @jridgewell to confirm.

Reply:
Yeah, that makes sense 👍